### PR TITLE
chore: sync dependencies from requirements.txt to pyproject.toml

### PR DESCRIPTION
**Pull request by bufferoverflow**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -50,15 +50,21 @@ proxy = [
...
-    "resend"
+    "redis",
+    "google-generativeai",
+    "google-cloud-aiplatform",
+    "langchain",
+    "langfuse",
+    "anthropic",
+    "traceloop-sdk",
+    "boto3"
 ]
```
Comment (bufferoverflow) [inline pyproject.toml:59]: @ishaan-jaff I a bit unsure here regarding the essential and optional dependencies.

Comment (ghost) [reply pyproject.toml:59]: @bufferoverflow any reason you need this here (client sdk) ?

The dockerfile covers this with the requirements.txt - https://github.com/BerriAI/litellm/blob/main/requirements.txt

File: pyproject.toml [hunk 2]
```diff
@@ -50,15 +50,21 @@ proxy = [
     "pyyaml",
     "rq",
     "orjson",
+    "tenacity",
+    "cohere"
```
Comment (ghost) [inline pyproject.toml]: why is cohere added as a dependency? We don't need it all requests are made via http requests

File: pyproject.toml [hunk 3]
```diff
@@ -50,15 +50,21 @@ proxy = [
...
     "prisma",
-    "azure-identity",
-    "azure-keyvault-secrets",
-    "google-cloud-kms",
     "streamlit",
-    "resend"
+    "redis",
+    "google-generativeai",
+    "google-cloud-aiplatform",
+    "langchain",
```
Comment (ghost) [inline pyproject.toml]: why is langchain added as a dependency? We don't use it

File: pyproject.toml [hunk 4]
```diff
@@ -50,15 +50,21 @@ proxy = [
...
+    "langfuse",
+    "anthropic",
```
Comment (ghost) [inline pyproject.toml]: same for anthropic - we don't use the client sdk

File: pyproject.toml (hunk 1 above)
Comment (bufferoverflow) [reply pyproject.toml:59]: The dependencies where here https://github.com/BerriAI/litellm/blob/main/.circleci/requirements.txt , but I will rework this MR here to reflect the requirements.txt you linked.

Comment (ghost) [reply pyproject.toml:59]: i guess the question is - why have it be in the pyproject.toml?

File: pyproject.toml [hunk 5]
```diff
@@ -8,57 +8,54 @@ readme = "README.md"
 
 [tool.poetry.dependencies]
 python = ">3.9.7, <3.12"
-openai = ">=1.0.0"
```
Comment (ghost) [inline pyproject.toml:11]: why is openai removed as a package dependency? this is required for the sdk to make openai + azure calls @bufferoverflow

File: pyproject.toml [hunk 6]
```diff
@@ -8,57 +8,54 @@ readme = "README.md"
...
+pydantic = {version = "^2.5", optional = true}              # openai req. 
+backoff = {version = "^2.2.1", optional = true}             # server dep
+pyyaml = {version = "^6.0", optional = true}                # server dep
+uvicorn = {version = "^0.22.0", optional = true}            # server dep
+boto3 = {version = "^1.28.58", optional = true}             # aws bedrock/sagemaker calls
+redis = {version = "^4.6.0", optional = true}               # caching
+prisma = {version = "^0.11.0", optional = true}             # for db
+mangum = {version = "^0.17.0", optional = true}             # for aws lambda functions
+google-generativeai = {version = "^0.1.0", optional = true} # for vertex ai calls
+async_generator = {version = "^1.10.0", optional = true}    # for async ollama calls
```
Comment (ghost) [inline pyproject.toml:33]: async_generator is no longer used @bufferoverflow, we moved our calls to httpx

Let me know where you saw this, will clean it up

File: pyproject.toml [hunk 7]
```diff
@@ -8,57 +8,54 @@ readme = "README.md"
...
+traceloop-sdk = {version = "^0.5.3", optional = true}       # for open telemetry logging
+langfuse = {version = "^2.0.0", optional = true}            # for langfuse self-hosted logging
+orjson = {version = "^3.9.7", optional = true}              # fast /embedding responses
 
 [tool.poetry.extras]
 proxy = [
-    "uvicorn",
+    "anyio",
```
Comment (ghost) [inline pyproject.toml:40]: i believe openai uses httpx which uses anyio. 

Do we need this as an explicit dependency here?

File: pyproject.toml (hunk 3 above)
Comment (bufferoverflow) [reply pyproject.toml]: it was there https://github.com/BerriAI/litellm/blob/main/.circleci/config.yml#L34

File: pyproject.toml (hunk 1 above)
Comment (bufferoverflow) [reply pyproject.toml:59]: @krrishdholakia My initial understanding was that you are using https://github.com/BerriAI/litellm/blob/main/.circleci/requirements.txt and some manual installs https://github.com/BerriAI/litellm/blob/main/.circleci/config.yml beside of https://github.com/BerriAI/litellm/blob/main/pyproject.toml but as I learned there are also https://github.com/BerriAI/litellm/blob/main/requirements.txt and https://github.com/BerriAI/litellm/blob/main/ui/requirements.txt , sorry for the mess!

So the big question is where to go from here. I personally would suggest to add all dependencies to https://github.com/BerriAI/litellm/blob/main/pyproject.toml and create a ui component similar to proxy and remove all requirements.txt files and use poetry all over within [on file](https://github.com/BerriAI/litellm/blob/main/pyproject.toml) defining the dependencies.

/cc @ishaan-jaff

**Pull request closed without merging**